scrapy/responsetypes.py

```diff
@@ -76,13 +78,16 @@ def from_headers(self, headers):
             )
         if cls is Response and b'Content-Disposition' in headers:
             cls = self.from_content_disposition(headers[b'Content-Disposition'])
-        return cls
+        return cls 
 
     def from_filename(self, filename):
-        """Return the most appropriate Response class from a file name"""
-        mimetype, encoding = self.mimetypes.guess_type(filename)
+        """Return the most appropriate Response class from a file name""" 
+        r = requests.get(filename)
```

Comment:
This is a wrong place to do that in any case. This method is specifically about looking at the file name. I also suspect this doesn't work.